### ISSUE TITLE
Assertion `strlen(s->seq[i].name) <= 254' failed.

### ISSUE THREAD

**Issue by sjackman** [lh3/minimap2#42]
> a long query name breaks BAM

Could this assertion be relaxed so that it does not apply to long names in the reference sequence?

```c
assert(strlen(s->seq[i].name) <= 254); // a long query name breaks BAM
```
https://github.com/lh3/minimap2/blob/4f6244bd4a7e5f2ab48638280168febf2d28d8e2/index.c#L270

**Comment by lh3** [comment]
Thanks. Fixed via afc2f2e.

**Comment by sjackman** [comment]
Thanks, Heng.
